plugins/openai_plugin.py

```diff
@@ -21,9 +22,22 @@
 """
 
 required_args = ["host", "streaming", "endpoint"]
+APIS = ["legacy", "chat"]
 
 logger = logging.getLogger("user")
 
+def deepget(obj: dict, *path, r: Any = None) -> Any:
+    """Acts like .get() but for nested objects."""
+    loc = obj
+    for p in path:
+        try:
+            loc = loc[p]
+        # NOTE: If loc is list then an invalid index throws IndexError
+        except (KeyError, IndexError):
+            return r
+    return loc
```

Comment:
can you clarify (illustrate) in the docstring how this function works?

in particular, I find the var name `r` not explicit. I undertstand that it's the default value, but that could be clarified

a common guideline is to avoid short variable names (`<= 3 chars`). Not an absolute rule, but it makes the code more readable.

Reply:
Ack

Reply:
🙏🏻